CMakeLists.txt

```diff
@@ -1,330 +1,712 @@
-
-endif ()
-
-if (BUILD_GUILE)
-  find_package (Guile)
-endif ()
-
-if (USE_SWIG)
-  find_package (SWIG)
-endif ()
-
-add_subdirectory (swig)
-
-if (BUILD_OCTAVE)
-  find_package (Octave)
-endif ()
-
-if (BUILD_MATLAB)
-  find_package (Matlab)
-endif ()
-
-if (OCTAVE_FOUND OR Matlab_FOUND)
-  add_subdirectory (octave)
-endif ()
-
-enable_testing ()
-add_subdirectory (test)
-
-set (CPACK_PACKAGE_NAME          "${CMAKE_PROJECT_NAME}")
-set (CPACK_PACKAGE_VERSION_MAJOR "${NLOPT_MAJOR_VERSION}")
-set (CPACK_PACKAGE_VERSION_MINOR "${NLOPT_MINOR_VERSION}")
-set (CPACK_PACKAGE_VERSION_PATCH "${NLOPT_BUGFIX_VERSION}")
-set (CPACK_SOURCE_GENERATOR      "TBZ2;TGZ"    )
-set (CPACK_BINARY_STGZ           "OFF"        )
-set (CPACK_BINARY_TBZ2           "ON"         )
-set (CPACK_BINARY_TGZ            "ON"         )
-set (CPACK_BINARY_TZ             "OFF"        )
-set (CPACK_SOURCE_IGNORE_FILES ".git;/build;.*~;${CPACK_SOURCE_IGNORE_FILES}")
-
-set (CPACK_SOURCE_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${NLOPT_VERSION_STRING})
-
-include (CPack)
-
-# configuration files
-export (TARGETS ${nlopt_lib} FILE ${PROJECT_BINARY_DIR}/NLoptLibraryDepends.cmake)
-
-# Install the export set for use with the install-tree
-install(EXPORT NLoptLibraryDepends
-        DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR}
-        COMPONENT Development)
-
-# Create a NLOPTConfig.cmake file for the use from the install tree
-# and install it
-set (NLOPT_LIBRARIES ${nlopt_lib})
-
-set (NLOPT_CMAKE_DIR "${INSTALL_CMAKE_DIR}")
-file (RELATIVE_PATH rel_include_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_INCLUDE_DIR}")
-list (APPEND RELATIVE_NLOPT_INCLUDE_DIRS ${rel_include_dir})
-
-file (RELATIVE_PATH rel_lib_dir "${NLOPT_CMAKE_DIR}" "${INSTALL_LIB_DIR}")
-list (APPEND RELATIVE_NLOPT_LIB_DIR ${rel_lib_dir})
-
-configure_file (cmake/NLoptConfig.cmake.in NLoptConfig.cmake @ONLY)
-configure_file (cmake/NLoptConfigVersion.cmake.in NLoptConfigVersion.cmake @ONLY)
-install (FILES
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfig.cmake
-          ${CMAKE_CURRENT_BINARY_DIR}/NLoptConfigVersion.cmake
-         DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR}
-         COMPONENT Development)
-
+#==============================================================================
+# NLOPT CMake file
+#
+# NLopt is a free/open-source library for nonlinear optimization, providing
+# a common interface for a number of different free optimization routines
+# available online as well as original implementations of various other
+# algorithms
+# WEBSITE: http://ab-initio.mit.edu/wiki/index.php/NLopt
+# AUTHOR: Steven G. Johnson
+#
+# This CMakeLists.txt file was created to compile NLOPT with the CMAKE utility.
+# Benoit Scherrer, 2010 CRL, Harvard Medical School
+# Copyright (c) 2008-2009 Children's Hospital Boston
+#==============================================================================
+
+cmake_minimum_required (VERSION 2.8.9)
+
+# default install directory locally inside build-tree
```

Comment:
I don't see the point of this.  If you're going to do `make install` at all, normally you would want it to be in a more globally visible directory, not inside the build tree — that's the whole point of `make install`.    The GNU standard default is `/usr/local`